server/client.go

```diff
@@ -447,6 +447,12 @@ func (c *client) processConnect(arg []byte) error {
 			srv.mu.Unlock()
 		}
 
+		// Check for max connections
```

Comment:
The client does not technically have to send a connect (think telnet to demo server). So we should probably check on accept and simply send an error there.